src/pretix/api/serializers/event.py

```diff
@@ -202,6 +202,87 @@ class Meta:
+
+        full_data = self.to_internal_value(self.to_representation(self.instance)) if self.instance else {}
+        full_data.update(data)
+
+        Event.clean_dates(data.get('date_from'), data.get('date_to'))
+        Event.clean_presale(data.get('presale_start'), data.get('presale_end'))
+
+        return data
+
+    @cached_property
+    def meta_properties(self):
+        return {
+            p.name: p for p in self.context['request'].organizer.meta_properties.all()
+        }
+
+    def validate_meta_data(self, value):
+        for key in value['meta_data'].keys():
+            if key not in self.meta_properties:
+                raise ValidationError(_('Meta data property \'{name}\' does not exist.').format(name=key))
+        return value
+
+    @transaction.atomic
+    def create(self, validated_data):
+        item_price_overrides_data = validated_data.pop('subeventitem_set') if 'subeventitem_set' in validated_data else {}
+        variation_price_overrides_data = validated_data.pop('subeventitemvariation_set') if 'subeventitemvariation_set' in validated_data else {}
+        meta_data = validated_data.pop('meta_data', None)
+        subevent = super().create(validated_data)
+
+        for item_price_override_data in item_price_overrides_data:
+            SubEventItem.objects.create(subevent=subevent, **item_price_override_data)
+        for variation_price_override_data in variation_price_overrides_data:
+            SubEventItemVariation.objects.create(subevent=subevent, **variation_price_override_data)
+
+        # Meta data
+        if meta_data is not None:
+            for key, value in meta_data.items():
+                subevent.meta_values.create(
+                    property=self.meta_properties.get(key),
+                    value=value
+                )
+
+        return subevent
+
+    @transaction.atomic
+    def update(self, instance, validated_data):
+        item_price_overrides_data = validated_data.pop('subeventitem_set') if 'subeventitem_set' in validated_data else {}
+        variation_price_overrides_data = validated_data.pop('subeventitemvariation_set') if 'subeventitemvariation_set' in validated_data else {}
+        meta_data = validated_data.pop('meta_data', None)
+        subevent = super().update(instance, validated_data)
+
+        for item_price_override_data in item_price_overrides_data:
+            item, created = SubEventItem.objects.get_or_create(subevent=subevent, item=item_price_override_data['item'])
+            item = SubEventItem(id=item.id, subevent=subevent, **item_price_override_data)
+            item.save()
+        for variation_price_override_data in variation_price_overrides_data:
+            variation, created = SubEventItemVariation.objects.get_or_create(subevent=subevent, variation=variation_price_override_data['variation'])
+            variation = SubEventItemVariation(id=variation.id, subevent=subevent, **variation_price_override_data)
+            variation.save()
+
```

Comment:
Remember deleting ones that have been present before but are no longer :) I recommend to fetch all existing SubEventItem(Variation)s before the loop into a dictionary mapping item IDs to SubEventItem objects, then doing the loop and .pop()-ing all objects from the dictionary that already exists, then deleting all that are still in the dictionary. This way, you can potentially save a database lookups at the same time!